src/helpers/transport/bytearrstream/mod.rs

```diff
@@ -0,0 +1,66 @@
+mod aligned;
+
+pub use aligned::ByteArrStream as AlignedByteArrStream;
+
+use crate::error::BoxError;
+use axum::extract::BodyStream;
+use futures::Stream;
+use futures_util::{stream, TryStreamExt};
+use hyper::body::Bytes;
+use pin_project::pin_project;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// represents the item of an underlying stream
+type Item = Result<Bytes, BoxError>;
+/// represents the underlying stream
+type PinnedStream = Pin<Box<dyn Stream<Item = Item> + Send>>;
+
+#[pin_project]
+pub struct ByteArrStream {
+    #[pin]
```

Comment:
do we need to pin here given that `PinnedStream` is pinned already?

Reply:
we don't, good catch